### PR TITLE
Bzrplugins

### PR DESCRIPTION
**Pull request by fmccann**
Adding additional formula for commonly used Bazaar plugins. These were created in the same fashion as the existing bzr-fastimport plugin.

### CODE REVIEW

**Comment by bfontaine** [comment]
We need an upstream release for `bzr-bisect`, `bzr-difftools` and `bzr-extmerge`. From the GitHub repos descriptions it appears you mirrored them and tagged a release yourself.

**Comment by fmccann** [comment]
Fair enough. I’ll check with the maintainers and see if we can get an official tarball. I assume there’s no way to pull from source control directly? I didn’t see it in the cookbook.

- Fred McCann
fred@sharpnoodles.com

**Comment by dunn** [comment]
> I assume there’s no way to pull from source control directly?

There is; `url "https://hostname/repo.git" will do a clone; but we only allow that for HEAD urls or for when cloning is required (e.g., to fetch submodules).  Otherwise we use the versioned tarballs.

**Comment by fmccann** [comment]
Gotcha. I’ve contacted the maintainers - we’re going to work on putting some official tarballs together.

- Fred McCann
fred@sharpnoodles.com

**Comment by fmccann** [comment]
I took a look at the docs again - it looks like using a VCS for the download strategy is an option. If I pull code from launchpad directly, would that be acceptable?

**Comment by dunn** [comment]
> If I pull code from launchpad directly, would that be acceptable?

Yeah, as long as there's a tag for a stable version we can use.

**Comment by fmccann** [comment]
OK, I've removed any formulae without stable tags, added tags for the rest.

File: Library/Formula/bzr-explorer.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class BzrExplorer < Formula
+  desc "Desktop application for using the Bazaar Version Control System"
+  homepage "https://launchpad.net/bzr-explorer"
+  url "lp:bzr-explorer", :using => :bzr, :tag => "release-1.3.0"
```
Comment (DomT4) [inline Library/Formula/bzr-explorer.rb]: Any reason we can't just use the tarball here?

File: Library/Formula/bzr-explorer.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
```
Comment (DomT4) [inline Library/Formula/bzr-explorer.rb]: Please add a `bottle :unneeded` line here if there's no compile stage, Thanks!

File: Library/Formula/bzr-explorer.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  def install
+    (share/"bazaar/plugins/explorer").install Dir["*"]
+  end
+
+  def caveats; <<-EOS.undent
+    In order to use this plugin you must set your PYTHONPATH in your ~/.bashrc:
+
+    export PYTHONPATH="#{HOMEBREW_PREFIX}/lib/python2.7/site-packages:$PYTHONPATH"
+
```
Comment (DomT4) [inline Library/Formula/bzr-explorer.rb]: Don't need this empty line.

File: Library/Formula/bzr-git.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+class BzrGit < Formula
+  desc "Git repository support for Bazaar"
+  homepage "https://launchpad.net/bzr-git"
+  url "lp:bzr-git", :using => :bzr, :tag => "release-0.6.10"
```
Comment (DomT4) [inline Library/Formula/bzr-git.rb]: Any reason we can't just use the tarball here?

File: Library/Formula/bzr-git.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  depends_on 'bazaar'
```
Comment (DomT4) [inline Library/Formula/bzr-git.rb]: Double quotes everywhere, Thanks!

File: Library/Formula/bzr-git.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    end
+
+    (share/"bazaar/plugins/git").install Dir["*"]
+  end
+
+  def caveats; <<-EOS.undent
+    In order to use this plugin you must set your PYTHONPATH in your ~/.bashrc:
+
+      export PYTHONPATH="#{opt_libexec}/vendor/lib/python2.7/site-packages:$PYTHONPATH"
+
```
Comment (DomT4) [inline Library/Formula/bzr-git.rb]: Same comment on spacing.

File: Library/Formula/bzr-xmloutput.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class BzrXmloutput < Formula
+  desc "Bazaar plugin that provides a option to generate XML output for builtins commands"
+  homepage "https://launchpad.net/bzr-xmloutput"
+  url "lp:bzr-xmloutput", :using => :bzr, :tag => "release-0.8.8"
```
Comment (DomT4) [inline Library/Formula/bzr-xmloutput.rb]: Any reason we can't just use the tarball here?

File: Library/Formula/bzr-xmloutput.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
```
Comment (DomT4) [inline Library/Formula/bzr-xmloutput.rb]: Same comment on `bottle :unneeded` and no compile.

File: Library/Formula/bzrtools.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+class Bzrtools < Formula
+  desc "Bazaar plugin that supplies useful additional utilities"
+  homepage "https://launchpad.net/bzrtools"
+  url "lp:bzrtools", :using => :bzr, :tag => "release-2.3.0"
```
Comment (DomT4) [inline Library/Formula/bzrtools.rb]: See above points on the tarball and bottle `:unneeded` :wink:

File: Library/Formula/qbzr.rb [hunk 10]
```diff
@@ -0,0 +1,24 @@
+class Qbzr < Formula
+  desc "Simple Qt cross-platform frontend for some of Bazaar commands"
+  homepage "https://launchpad.net/qbzr"
+  url "lp:qbzr", :using => :bzr, :tag => "release-0.23.1"
```
Comment (DomT4) [inline Library/Formula/qbzr.rb]: See above points on the tarball and bottle `:unneeded` :wink:

File: Library/Formula/qbzr.rb [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+  def install
+    (share/"bazaar/plugins/qbzr").install Dir["*"]
+  end
+
+  def caveats; <<-EOS.undent
+    In order to use this plugin you must set your PYTHONPATH in your ~/.bashrc:
+
+    export PYTHONPATH="#{HOMEBREW_PREFIX}/lib/python2.7/site-packages:$PYTHONPATH"
+
```
Comment (DomT4) [inline Library/Formula/qbzr.rb]: And caveat spacing.

**Comment by fmccann** [comment]
I'm going to break this up into a pull request per plugin. I think that will make it easier manage.

**Comment by UniqMartin** [comment]
@fmccann Breaking this up is a good idea, but it would be much appreciated if you could stick with the `<formula-name> <version> (new formula)` subject format for both commit messages and PRs. Makes things easier for us …

**Comment by fmccann** [comment]
Will do!

- Fred McCann
fred@sharpnoodles.com

**Pull request closed without merging**